azure-iot-device/tests/iothub/pipeline/test_pipeline_stages_iothub.py

```diff
@@ -65,6 +69,8 @@ def init_kwargs(self):
     @pytest.fixture
     def stage(self, mocker, cls_type, init_kwargs):
         stage = cls_type(**init_kwargs)
+        stage.nucleus = pipeline_nucleus.PipelineNucleus(pipeline_configuration=mocker.MagicMock())
```

Comment:
This change will allow tests to pass, but you now don't have correct coverage. Adding this kwarg introduced functionality differences and not only do we have to write tests to cover these differences, but we have to make sure no previously written tests are too tied to the old paradigm.

This might be a little tricky - we can work on it together on Monday perhaps?